docs/src/arrays.md

```diff
@@ -0,0 +1,219 @@
+
+## Forward mode rules
+
+Given a function
+
+```julia
+C = f(A::Array, B::Array)::Array
+```
+
+we write the differential of the output in terms of differentials of the inputs as
+
+$$dC = \frac{\partial f}{\partial A} dA + \frac{\partial f}{\partial B} dB.$$
+
+The pushforward is obtained by dividing by the differential wrt a single scalar input $t$.
+Writing $\dot{C} = \frac{dC}{dt}$, we replace the differentials with the forward mode sensitivities:
+
+$$\dot{C} = \frac{\partial f}{\partial A} \dot{A} + \frac{\partial f}{\partial B} \dot{B}.$$
+
+The terms $\frac{\partial f}{\partial A}$ are array-array derivatives, i.e. a type of Jacobian.
+We do not write these down explicitly, but we instead use differential identities to write down the terms $\frac{\partial f}{\partial A} dA$, which as we've seen behave like the Jacobian-vector-products $\frac{\partial f}{\partial A} \dot{A}$.
+The differential identities follow directly from the usual scalar identities.
+
+### Matrix addition
+
+$$C = A + B$$
+
+$$dC = dA + dB$$
+
+### Matrix multiplication
+
+$$C = A B$$
+
+First we write in component form.
+
+$$C_{ij} = \sum_k A_{ik} B_{kj}$$
+
+Then we apply the product rule to get the scalar differential identity.
+
+$$dC_{ij} = \sum_k dA_{ik}\ B_{kj} + A_{ik}\ dB_{kj}$$
+
+But this is just a sum matrix products:
+
+$$C = dA\ B + A\ dB$$
+
+So we now have the matrix product rule.
+
+### Matrix inversion
+
+$$C A = A C = I$$
+
+Here we differentiate the constraint, applying the matrix product rule, and solve for $dC$.
+
+$$dC\ A + C\ dA = 0$$
+$$dC\ A\ C + C\ dA\ C = 0$$
+$$dC = -C\ dA\ C$$
+
+### Other useful identities
+
+Two useful identities are $d(A^H) = dA^H$ and $d(A^T) = dA^T$, where $\cdot^H$ is the conjugate transpose.
+
+## Writing reverse-mode rules
+
+Reverse-mode rules are a little more involved.
+For a real scalar function $s = g(C)$, the differential of $s$ is the real part of the inner product of the adjoint of $C$, $\overline{C}$, and the differential of $C$:
+
+$$ds = \Re\langle \overline{C}, dC \rangle = \Re\left(\sum_{i,\dots,j} \overline{C}_{i,\dots,j}^* dC_{i,\dots,j}\right),$$
+
+where $\cdot^*$ is the complex conjugate, and $\Re(x)$ is the real part of $x$ (i.e. `real(x)`).
+
+For matrices and vectors, we can write this as
+
+$$ds = \Re(\operatorname{tr}(\overline{C}^H\ dC)),$$
+
+where $\operatorname{tr}$ is the `trace`.
+
+Plugging in the identity for $dC$, we get
+
+$$ds = \Re\left(\operatorname{tr}\left(\overline{C}^H \frac{\partial f}{\partial A} dA + \overline{C}^H\frac{\partial f}{\partial B} dB\right)\right) = \Re\left(\operatorname{tr}\left(\overline{C}^H \frac{\partial f}{\partial A} dA\right)\right) + \Re\left(\operatorname{tr}\left(\overline{C}^H\frac{\partial f}{\partial B} dB\right)\right)$$
+
+By applying the same definition of $ds$ to the intermediates $A$ and $B$, we get
+
+$$ds = \Re\left(\operatorname{tr}(\overline{A}^H\ dA)\right) + \Re\left(\operatorname{tr}(\overline{B}^H\ dB)\right)$$
```

Comment:
I think another line here that explicitly highlights the fact that you've equated `\overline{A}^H` and ``\overline{C}^H df / dA` would be helpful. There's a lot of notation flying around here, so it might well take someone who's new to this a little while to figure out that you're just equating terms.